File: csharp/BlackJack/Models/Deck.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace BlackJack
{
    public class Deck
    {
        public Queue<Card> Cards;

        private List<Card> UnshuffledCards;
        public Deck()
        {
            UnshuffledCards = new List<Card>();
            foreach (Suit suit in Enum.GetValues(typeof(Suit)))
            {
                foreach (CardRank rank in Enum.GetValues(typeof(CardRank)))
                {
                    UnshuffledCards.Add(new Card() { Rank = rank, Suit = suit });
                }
            }
            var rnd = new Random();
            var randomized = UnshuffledCards.OrderBy(card => rnd.Next());
            Cards = new Queue<Card>();
            foreach (Card card in randomized)
            {
                Cards.Enqueue(card);
            }
        }
    }

}

File: typescript/src/deck.ts
import { Card } from "./card";
import { Suit } from "./suit";
export class Deck {
  public cards: Array<Card> = [];

  constructor() {
    for (var suit in Object.keys(Suit)) {
      for (var rank = 1; rank < 14; rank++) {
        this.cards.push(new Card(rank, Object.values(Suit)[suit]));
      }
    }
  }
}
